google-cloud-firestore/src/main/java/com/google/cloud/firestore/Order.java

```diff
@@ -65,13 +67,27 @@ static TypeOrder fromValue(Value value) {
         case ARRAY_VALUE:
           return ARRAY;
         case MAP_VALUE:
-          return OBJECT;
+          return fromMapValue(value);
```

Comment:
I'm not sure I fully understand the concern. The sort order implemented here aligns with the backend sort order of vectors and maps. It's unfortunate that in older SDKs vectors will sort in the middle of other map values, but I don't think this will cause any issues.

Reply:
Yes ... my concern only applies to old SDKs.